File: src/DBHandler/DbProvider/ParameterManager.cs
using MySql.Data.MySqlClient;
using System.Data;
using System.Data.SqlClient;
using System.Data.SQLite;

namespace DBHandler.DbProvider
{
    /// <summary>
    /// Author : Faris Jaafar
    /// Created Date : FEB 2019
    /// UPDATED : -
    /// VERSION : 1.0
    /// </summary>
    public sealed class ParameterManager
    {
        public static IDbDataParameter CreateParameter(string providerName, string name, object value, DbType dbType, ParameterDirection direction = ParameterDirection.Input)
        {
            IDbDataParameter parameter = null;
            switch (providerName.ToLower())
            {
                case "system.data.sqlclient":
                    return CreateSqlParameter(name, value, dbType, direction);
                case "mysql.data.mysqlclient":
                    return CreateMySqlParameter(name, value, dbType, direction);
                case "system.data.sqlite":
                    return CreateSQLiteParameter(name, value, dbType, direction);
                case "system.data.oleDb":
                    break;
                case "system.data.odbc":
                    break;
            }

            return parameter;
        }

        public static IDbDataParameter CreateParameter(string providerName, string name, int size, object value, DbType dbType, ParameterDirection direction = ParameterDirection.Input)
        {
            IDbDataParameter parameter = null;
            switch (providerName.ToLower())
            {
                case "system.data.sqlclient":
                    return CreateSqlParameter(name, size, value, dbType, direction);
                case "mysql.data.mysqlclient":
                    return CreateMySqlParameter(name, size, value, dbType, direction);
                case "system.data.sqlite":
                    return CreateSQLiteParameter(name, size, value, dbType, direction);
                case "system.data.oleDb":
                    break;
                case "system.data.odbc":
                    break;
            }

            return parameter;
        }

        private static IDbDataParameter CreateSqlParameter(string name, object value, DbType dbType, ParameterDirection direction)
        {
            return new SqlParameter
            {
                DbType = dbType,
                ParameterName = name,
                Direction = direction,
                Value = value
            };
        }

        private static IDbDataParameter CreateSqlParameter(string name, int size, object value, DbType dbType, ParameterDirection direction)
        {
            return new SqlParameter
            {
                DbType = dbType,
                Size = size,
                ParameterName = name,
                Direction = direction,
                Value = value
            };
        }

        private static IDbDataParameter CreateMySqlParameter(string name, object value, DbType dbType, ParameterDirection direction)
        {
            return new MySqlParameter
            {
                DbType = dbType,
                ParameterName = name,
                Direction = direction,
                Value = value
            };
        }

        private static IDbDataParameter CreateMySqlParameter(string name, int size, object value, DbType dbType, ParameterDirection direction)
        {
            return new MySqlParameter
            {
                DbType = dbType,
                Size = size,
                ParameterName = name,
                Direction = direction,
                Value = value
            };
        }

        private static IDbDataParameter CreateSQLiteParameter(string name, object value, DbType dbType, ParameterDirection direction)
        {
            return new SQLiteParameter
            {
                DbType = dbType,
                ParameterName = name,
                Direction = direction,
                Value = value
            };
        }

        private static IDbDataParameter CreateSQLiteParameter(string name, int size, object value, DbType dbType, ParameterDirection direction)
        {
            return new SQLiteParameter
            {
                DbType = dbType,
                Size = size,
                ParameterName = name,
                Direction = direction,
                Value = value
            };
        }
        //private static IDbDataParameter CreateOracleParameter(string name, object value, DbType dbType, ParameterDirection direction)
        //{
        //    return new OracleParameter
        //    {
        //        DbType = dbType,
        //        ParameterName = name,
        //        Direction = direction,
        //        Value = value
        //    };
        //}

        //private static IDbDataParameter CreateOracleParameter(string name, int size, object value, DbType dbType, ParameterDirection direction)
        //{
        //    return new OracleParameter
        //    {
        //        DbType = dbType,
        //        Size = size,
        //        ParameterName = name,
        //        Direction = direction,
        //        Value = value
        //    };
        //}
    }
}

File: src/DBHandlerVB/DbProvider/ParameterManager.vb
Imports System.Data.SqlClient
Imports System.Data.SQLite
Imports MySql.Data.MySqlClient

Namespace DBHandler.DbProvider
    Public Class ParameterManager
        Public Shared Function CreateParameter(ByVal providerName As String, ByVal name As String, ByVal value As Object, ByVal dbType As DbType, ByVal Optional direction As ParameterDirection = ParameterDirection.Input) As IDbDataParameter
            Dim parameter As IDbDataParameter = Nothing

            Select Case providerName.ToLower()
                Case "system.data.sqlclient"
                    Return CreateSqlParameter(name, value, dbType, direction)
                Case "mysql.data.mysqlclient"
                    Return CreateMySqlParameter(name, value, dbType, direction)
                Case "system.data.sqlite"
                    Return CreateSQLiteParameter(name, value, dbType, direction)
                Case "system.data.oleDb"
                Case "system.data.odbc"
            End Select

            Return parameter
        End Function

        Public Shared Function CreateParameter(ByVal providerName As String, ByVal name As String, ByVal size As Integer, ByVal value As Object, ByVal dbType As DbType, ByVal Optional direction As ParameterDirection = ParameterDirection.Input) As IDbDataParameter
            Dim parameter As IDbDataParameter = Nothing

            Select Case providerName.ToLower()
                Case "system.data.sqlclient"
                    Return CreateSqlParameter(name, size, value, dbType, direction)
                Case "mysql.data.mysqlclient"
                    Return CreateMySqlParameter(name, size, value, dbType, direction)
                Case "system.data.sqlite"
                    Return CreateSQLiteParameter(name, size, value, dbType, direction)
                Case "system.data.oleDb"
                Case "system.data.odbc"
            End Select

            Return parameter
        End Function

        Private Shared Function CreateSqlParameter(ByVal name As String, ByVal value As Object, ByVal dbType As DbType, ByVal direction As ParameterDirection) As IDbDataParameter
            Return New SqlParameter With {
                .DbType = dbType,
                .ParameterName = name,
                .Direction = direction,
                .Value = value
            }
        End Function

        Private Shared Function CreateSqlParameter(ByVal name As String, ByVal size As Integer, ByVal value As Object, ByVal dbType As DbType, ByVal direction As ParameterDirection) As IDbDataParameter
            Return New SqlParameter With {
                .DbType = dbType,
                .Size = size,
                .ParameterName = name,
                .Direction = direction,
                .Value = value
            }
        End Function

        Private Shared Function CreateMySqlParameter(ByVal name As String, ByVal value As Object, ByVal dbType As DbType, ByVal direction As ParameterDirection) As IDbDataParameter
            Return New MySqlParameter With {
                .DbType = dbType,
                .ParameterName = name,
                .Direction = direction,
                .Value = value
            }
        End Function

        Private Shared Function CreateMySqlParameter(ByVal name As String, ByVal size As Integer, ByVal value As Object, ByVal dbType As DbType, ByVal direction As ParameterDirection) As IDbDataParameter
            Return New MySqlParameter With {
                .DbType = dbType,
                .Size = size,
                .ParameterName = name,
                .Direction = direction,
                .Value = value
            }
        End Function

        Private Shared Function CreateSQLiteParameter(ByVal name As String, ByVal value As Object, ByVal dbType As DbType, ByVal direction As ParameterDirection) As IDbDataParameter
            Return New SQLiteParameter With {
                .DbType = dbType,
                .ParameterName = name,
                .Direction = direction,
                .Value = value
            }
        End Function
        Private Shared Function CreateSQLiteParameter(ByVal name As String, ByVal size As Integer, ByVal value As Object, ByVal dbType As DbType, ByVal direction As ParameterDirection) As IDbDataParameter
            Return New SQLiteParameter With {
                .DbType = dbType,
                .Size = size,
                .ParameterName = name,
                .Direction = direction,
                .Value = value
            }
        End Function
    End Class
End Namespace
